libc/string_errsig.c

```diff
@@ -0,0 +1,312 @@
+#define MAX_LEN_STRING 60
+
+/* ECANCELED, EDQUOT, EIDRM, EMULTIHOP, ENOLINK, ENOSR, ENOSTR, EOWNERDEAD, ESTALE missing in Phoenix-RTOS
+ *  #689 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/689
+ */
+static const int error_codes[] = { E2BIG, EACCES, EADDRINUSE, EADDRNOTAVAIL, EAFNOSUPPORT, EAGAIN, EALREADY, EBADF,
+	EBADMSG, EBUSY, ECHILD, ECONNABORTED, ECONNREFUSED, ECONNRESET, EDEADLK, EDESTADDRREQ, EDOM, EEXIST, EFAULT,
+	EFBIG, EHOSTUNREACH, EILSEQ, EINPROGRESS, EINTR, EINVAL, EIO, EISCONN, EISDIR, ELOOP, EMFILE, EMLINK, EMSGSIZE,
+	ENAMETOOLONG, ENETDOWN, ENETRESET, ENETUNREACH, ENFILE, ENOBUFS, ENODATA, ENODEV, ENOENT, ENOEXEC, ENOLCK, ENOMEM,
+	ENOMSG, ENOPROTOOPT, ENOSPC, ENOSYS, ENOTCONN, ENOTDIR, ENOTEMPTY, ENOTRECOVERABLE, ENOTSOCK, ENOTSUP, ENOTTY,
+	ENXIO, EOPNOTSUPP, EOVERFLOW, EPERM, EPIPE, EPROTO, EPROTONOSUPPORT, EPROTOTYPE, ERANGE, EROFS, ESPIPE, ESRCH,
+	ETIME, ETIMEDOUT, ETXTBSY, EWOULDBLOCK, EXDEV, ENOTBLK };
+
+static const int signal_codes[] = { SIGABRT, SIGALRM, SIGBUS, SIGCHLD, SIGCONT, SIGFPE, SIGHUP, SIGILL, SIGINT,
+	SIGKILL, SIGPIPE, SIGQUIT, SIGSEGV, SIGSTOP, SIGTERM, SIGTSTP, SIGTTIN, SIGTTOU, SIGUSR1, SIGUSR2, SIGPROF,
+	SIGSYS, SIGTRAP, SIGURG, SIGVTALRM, SIGXCPU, SIGXFSZ };
+
+static const unsigned int error_codes_len = sizeof(error_codes) / sizeof(error_codes[0]);
+
+static const unsigned int signal_codes_len = sizeof(signal_codes) / sizeof(signal_codes[0]);
+
+TEST_GROUP(string_errsign);
+
+
+TEST_SETUP(string_errsign)
+{
+}
+
+
+TEST_TEAR_DOWN(string_errsign)
+{
+}
+
+
+TEST(string_errsign, strerror_basic)
+{
+	for (int i = 0; i < error_codes_len; i++) {
+		errno = 0;
+		TEST_ASSERT_NOT_NULL(strerror(error_codes[i]));
+
+		if (i != 0) {
+			TEST_ASSERT_NOT_NULL(strerror(error_codes[i - 1]));
+			TEST_ASSERT_NOT_EQUAL_INT(0, strcmp(strerror(error_codes[i - 1]), strerror(error_codes[i])));
+		}
+
+		TEST_ASSERT_EQUAL_INT(0, errno);
+	}
+}
+
+
+TEST(string_errsign, strerror_zero)
+{
+	errno = 0;
+	TEST_ASSERT_NOT_NULL(strerror(0));
+	TEST_ASSERT_EQUAL_INT(0, errno);
+}
+
+
+TEST(string_errsign, strerror_invaild)
+{
+	const int values[] = { INT_MIN, INT_MIN / 2, INT_MIN / 4, -1024, -256, 256, 1024, INT_MAX / 4, INT_MAX / 2, INT_MAX };
+	for (int i = 0; i < (sizeof(values) / sizeof(values[0])); i++) {
+		TEST_ASSERT_NOT_NULL(strerror(values[i]));
+		TEST_ASSERT_NOT_NULL(strstr(strerror(values[i]), "Unknown error"));
+	}
+}
+
+
+TEST(string_errsign, strerror_r_basic)
+{
+/* Disabled because of #694 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/694 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#694 issue");
+#else
+	for (int i = 0; i < error_codes_len; i++) {
+		char prev_buf[MAX_LEN_STRING] = { 0 };
+		char buf[MAX_LEN_STRING] = { 0 };
+
+		errno = 0;
+		TEST_ASSERT_EQUAL_INT(0, strerror_r(error_codes[i], buf, sizeof(buf)));
+
+		if (i != 0) {
+			TEST_ASSERT_EQUAL_INT(0, strerror_r(error_codes[i - 1], prev_buf, sizeof(prev_buf)));
```

Comment:
I suppose you wanted to do:
```suggestion
			TEST_ASSERT_EQUAL_INT(0, strerror_r(error_codes[i - 1], buf, sizeof(buf)));
```

Reply:
This is for check if the previous message is different from the currently checked one